events/events_network.c

```diff
@@ -136,6 +136,15 @@ events_network_register(int (*func)(void *), void * cookie, int s, int op)
 		goto err0;
 
 	/*
+	 * There is a new socket/operation pair:
+	 * Rebuild the fd list on the next call to events_network_select().
+	 */
+	if (fds != NULL) {
```

Comment:
Yes, the fd sets were being cleared for every call to select.  But not for every register/deregister call.  The scenario I'm concerned about here is
1. select gets called (once)
2. 100 active connections get callbacks to read/write data and register new callbacks.